src/uhs/twophase/sentinel_2pc/controller.cpp

```diff
@@ -25,6 +25,12 @@ namespace cbdc::sentinel_2pc {
                                                    .size())]) {}
 
     auto controller::init() -> bool {
+        if(m_sentinel_id > (m_opts.m_sentinel_endpoints.size() - 1)) {
```

Comment:
Does this check still work if `m_sentinel_endpoints.size()` is zero? Does `m_opts.m_sentinel_endpoints.size() - 1` result in underflow, bypassing the check?